src/core_inner/cluster/quorum.rs

```diff
@@ -305,6 +324,32 @@ impl Cluster for Quorum {
         .boxed();
         BackendGet(task)
     }
+
+    fn exist_clustered_async(&self, keys: &[BobKey]) -> Exist {
+        let keys_by_nodes = self.group_keys_by_nodes(keys);
+        debug!(
+            "EXIST Nodes for fan out: {:?}",
+            print_vec(&keys_by_nodes.keys().flatten().collect::<Vec<_>>())
+        );
+        let len = keys.len();
+        Exist(
+            async move {
+                let mut exist = vec![false; len];
+                for (nodes, (keys, indexes)) in keys_by_nodes {
+                    let cluster_results = LinkManager::exist_on_nodes(&nodes, keys).await;
+                    for result in cluster_results {
+                        if let Ok(result) = result {
+                            for (&r, &ind) in result.result.exist.iter().zip(&indexes) {
```

Comment:
Triple nesting of `for` cycle is a nightmare :laughing: , but I don't know if it can be done in better way in this case. Maybe we'll leave it as it is until further refactoring